### PR TITLE
Improvement/get child organizations ids with recursive

### PR DESCRIPTION
**Pull request by BimsaraBodaragama**
Resolves: [Issue-21753](https://github.com/wso2/product-is/issues/21753)

## Automation tests
 - Unit tests 
   > Verified recursive and non-recursive behavior through new test cases.

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/OrganizationManagementDAO.java [hunk 1]
```diff
@@ -221,13 +221,26 @@ boolean isAttributeExistByKey(String organizationId, String attributeKey)
...
+    /**
+     * Retrieve the list of child organization IDs of a given organization.
+     *
+     * @param organizationId The organization ID.
+     * @param recursive      Determines whether records should be retrieved in a recursive manner.
+     * @return the ID list of the child organizations.
+     * @throws OrganizationManagementServerException The server exception thrown when retrieving the child
+     *                                               organizations.
+     */
+    List<String> getChildOrganizationIds(String organizationId, boolean recursive)
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/OrganizationManagementDAO.java:241]: make this a default method

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java [hunk 2]
```diff
@@ -245,6 +245,11 @@ public class SQLConstants {
             "WHERE UM_ORG_HIERARCHY.UM_PARENT_ID = :" + SQLPlaceholders.DB_SCHEMA_COLUMN_NAME_PARENT_ID +
             "; AND UM_ORG_HIERARCHY.DEPTH %s";
 
+    public static final String GET_CHILD_ORGANIZATION_IDS_WITH_RECURSIVE = "SELECT UM_ORG.UM_ID FROM UM_ORG " +
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java:248]:
```suggestion
    public static final String GET_CHILD_ORGANIZATION_IDS = "SELECT UM_ORG.UM_ID FROM UM_ORG " +
```

Comment (BimsaraBodaragama) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java:248]: We already have a variable named GET_CHILD_ORGANIZATION_IDS

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/OrganizationManagementDAO.java (hunk 1 above)
Comment (BimsaraBodaragama) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/OrganizationManagementDAO.java]: Resolved with e934897

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java [hunk 3]
```diff
@@ -108,6 +108,17 @@ Organization getOrganization(String organizationId, boolean showChildren, boolea
...
 
+    /**
+     * Returns the unique identifiers of the child organizations for a given organization.
+     *
+     * @param organizationId The organization ID.
+     * @param recursive      Determines whether records should be retrieved in a recursive manner.
+     * @return the list of Child organization IDs.
+     * @throws OrganizationManagementException exception is thrown when listing organizations.
+     */
+    List<String> getChildOrganizationsIds(String organizationId, boolean recursive)
```
[the remark this answers is not in the document]
Comment (BimsaraBodaragama) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java:119]: Resolved with e934897

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java (hunk 2 above)
Comment (BimsaraBodaragama) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java]: Used GET_IMMEDIATE_OR_ALL_CHILD_ORG_IDS instead
Resolved with e934897

**Review by AnuradhaSK — DISMISSED** [review]

**Review by AnuradhaSK — APPROVED** [review]

File: components/org.wso2.carbon.identity.organization.management.service/src/test/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImplTest.java [hunk 4]
```diff
@@ -701,6 +701,28 @@ public void testResolveTenantDomainWithEmptyOrganizationId() throws Organization
 
         organizationManager.resolveTenantDomain("");
     }
+  
+    @Test
+    public void testGetChildOrganizationIds() throws Exception {
+
+        TestUtils.mockCarbonContext(SUPER_ORG_ID);
+
+        // Non-recursive test case (only direct children)
```
Comment (sahandilshan) [inline components/org.wso2.carbon.identity.organization.management.service/src/test/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImplTest.java:710]: Add a full stop at the end of the sentence. Check other places

**Pull request merged**